GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -848,6 +871,92 @@ private String optionCellValue(String value) {
         return optionString.toString();
     }
 
+    /*
+     * Creates list of option values. The first value is always the pipe-separated format
+     * Depending on the useQuestionId parameter, each option is given its own column
+     * A 0 or 1 denotes if that option was selected or not
+     * if the AllowOther flag is true, a column is created for the Other option.
+     *
+     * We first try to match on text, and if that fails, we try to match on code. This
+     * guards against texts that are slightly changed during the evolution of a survey
+     */
+    private List<String> optionCellValues(Long questionId, String value, boolean useQuestionId, List<QuestionOptionDto> options, Boolean allowOther) {
+    	List<String> cells = new ArrayList<>();
+
+    	// get optionNodes from packed string value
+    	List<Map<String,String>> optionNodes = getNodes(value);
+
+    	// build pipe-separated format and add this to cell list
+    	String optionString = buildOptionString(optionNodes);
+        cells.add(optionString);
+
+        // if needed, build cells for options
+        if (useQuestionId){
+        	String text;
+        	String code;
+        	String cacheId;
+        	String other = null;
+        	boolean found;
+        	int numOptions = options.size();
+        	boolean[] optionFound = new boolean[numOptions];
+        	String qId = questionId.toString();
+
+        	for (Map<String,String> optAnswer : optionNodes){
+        		text = optAnswer.get("text") != null ? optAnswer.get("text") : "";
+        		code = optAnswer.get("code") != null ? optAnswer.get("code") : "";
+        		found = false;
+
+        		// try cache first
+        		cacheId = qId + text + code;
+        		if (optionCache.containsKey(cacheId)){
+        			optionFound[optionCache.get(cacheId)] = true;
+        			found = true;
+        		}
+
+        		// If it is not in the cache, try to match on text
+        		if (!found){
+	        		for (int i = 0; i < numOptions; i++){
+	            		if (text != null && text.equalsIgnoreCase(options.get(i).getText()) && text.length() > 0){
+	            			optionFound[i] = true;
+	            			found = true;
+	            			// put in cache
+	            			optionCache.put(cacheId, i);
+	            			break;
+	            		}
+	            	}
+        		}
+
+        		// finally, try to match on code
```

Comment:
Perhaps we should try to match by code first and if that fails then match by text.  The current setup is such that there will always be text but the codes may or may not be set.  Also the codes are all or nothing so if there is no code for the first option we don't bother checking for codes for the rest of the responses.